pkg/controller/keystone/keystone_controller.go

```diff
@@ -238,6 +244,43 @@ func (r *ReconcileKeystone) Reconcile(request reconcile.Request) (reconcile.Resu
 	return reconcile.Result{}, r.updateStatus(keystone, sts, keystoneClusterIP)
 }
 
+func (r *ReconcileKeystone) setDefaultValues(k *contrail.Keystone) error {
+	patch := client.MergeFrom(k.DeepCopy())
```

Comment:
Maybe this patch could be moved closer to r.client.Patch

Reply:
Moved this part to manager. Should be done just once, and not checked constantly at runtime.